### PR TITLE
Allow LspWorkspaceSymbol query on cmdline

### PR DESCRIPTION
**Pull request by idbrii**
**Caveat**: I haven't tested this with an lsp-server because none of my lsp (pyls, clangd) support it. However, I've tested that l:query seems to be filled with an appropriate value (disable return and echo `l:query`).

After investigating #741, I found that LspWorkspaceSymbol's use of inputdialog() makes it not support pasting from registers. This provides an alternative (and a way I'd prefer to use it).

Allow users to specify their query on the cmdline so they have easier
access to registers and vim-editing methods (especially with ctrl-f to
enter cmdline-window).

### CODE REVIEW

File: autoload/lsp/ui/vim.vim [hunk 1]
```diff
@@ -343,9 +343,13 @@ function! lsp#ui#vim#workspace_symbol() abort
         return
     endif
 
-    let l:query = inputdialog('query>', '', "\<ESC>")
-    if l:query ==# "\<ESC>"
-        return
+    if len(a:query) > 0
```
Comment (mattn) [inline autoload/lsp/ui/vim.vim]:
```
if empty(a:query)
```

Comment (idbrii) [reply autoload/lsp/ui/vim.vim]: I've made this change, but there's a lot of len() == 0 checks in the code. Is empty() new and using it is possibly a backwards-compatibility breaker? Github can't handle showing me blame for when it was added to vim/vim.

Comment (idbrii) [reply autoload/lsp/ui/vim.vim]: empty() existed since 7.0034.

**Comment by idbrii** [comment]
Rebased on top of master.

**Comment by prabirshrestha** [comment]
Tried it out and it works. You can also use `ctrl+r` to paste from registry.

Good part about this change is it will allow to easily use `<cword>` and other extensions to make it easy to fill.

**Pull request merged**